kif-framework/KIF
# Carthage

Implements [KIF/KIF#552](https://github.com/kif-framework/KIF/issues/552).

To verify:

#1: Grab this KIF fork and run unit tests:

````
git clone https://github.com/lottadot/KIF.git
cd KIF
git checkout carthage
xcodebuild test -scheme KIFFramework -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 6,OS=latest'
````

#2: Grab example project:

````
brew install Carthage
git clone https://github.com/lottadot/KIF-Carthage-Example.git
cd KIF-Carthage-Example
carthage checkout
xcrun xcodebuild test -workspace KIF-Carthage-Example.xcworkspace -scheme ExampleAcceptanceTests -sdk iphonesimulator -destination 'platform=iOS Simulator,name=iPhone 6,OS=latest'
````

You’ll note that we aren’t doing `carthage bootstrap`. That’s intentional (for now). We cannot yet build KIFFramework via Carthage that is app-store Submittable (iOS8 Dynamic Frameworks must be code signed, and that is where Carthage has complications in trying to determine when to (not)sign, what to strip out of the binary, when, etc). 

I realize that most people do not submit to the App store w/ a build target that has KIF in it (hopefully). However, if we try to use the release version of Carthage and build we are shown code-siging issues (which Carthage has issues discussing too). See note in [KIF-Carthage-Example-README](https://github.com/lottadot/KIF-Carthage-Example/blob/master/README.md)

Here's the result of that, for the record:

````
git clone https://github.com/lottadot/KIF-Carthage-Example.git
cd KIF-Carthage-Example
$ carthage bootstrap --verbose
*** No Cartfile.resolved found, updating dependencies
*** Fetching KIF
*** Checking out KIF at "e66d724a01cc0b5911b5b583f894e478df8cb1e7"
*** Building scheme "KIFFramework" in KIF.xcodeproj
Build settings from command line:
    SDKROOT = iphonesimulator8.3
....
=== BUILD TARGET KIFFrameworkTests OF PROJECT KIF WITH CONFIGURATION Release ===

Check dependencies
`CodeSign error: code signing is required for product type 'Unit Test Bundle' in SDK 'iOS 8.3'`

2015-04-12 11:25:06.244 xcodebuild[17168:379933]  DeveloperPortal: Using pre-existing current store at URL (DeveloperPortal%206.3.db).
** BUILD FAILED **
````
